src/verilog/verilog_expr.cpp

```diff
@@ -81,6 +81,34 @@ void verilog_module_sourcet::show(std::ostream &out) const
   out << '\n';
 }
 
+static void
+dependencies_rec(const verilog_module_itemt &, std::vector<irep_idt> &);
+
+static void dependencies_rec(const exprt &expr, std::vector<irep_idt> &dest)
+{
+  if(expr.id() == ID_verilog_package_scope)
+  {
+    auto &package_scope_expr = to_verilog_package_scope_expr(expr);
+    dest.push_back(
+      verilog_package_identifier(package_scope_expr.package_base_name()));
+  }
+  else
+  {
+    for(auto &op : expr.operands())
+      dependencies_rec(op, dest);
+  }
```

Comment:
In the interest of reducing (or: not further increasing) the use of recursion when traversing expressions (which at times is causing stack overflows for us): can we please use an expr iterator here:
````C++
for(const_depth_iteratort it = expr.depth_cbegin(); it != expr.depth_cend(); ++it)
{
  if(it->id() == ID_verilog_package_scope)
  {
    auto &package_scope_expr = to_verilog_package_scope_expr(*it);
    dest.push_back(
      verilog_package_identifier(package_scope_expr.package_base_name()));
  }
}
````